core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -792,6 +793,11 @@ private[spark] object Utils extends Logging {
   lazy val localIpAddress: String = findLocalIpAddress()
   lazy val localIpAddressHostname: String = getAddressHostName(localIpAddress)
 
+  def localIpURI(): String = {
```

Comment:
`toUriString` takes in an `InetAddress`. If I do the formatting at the caller, I'd need to copy paste two lines of code in two places. I was hoping to avoid that by putting it in Utils. However, in the last patch, I did miss some logic around handling env variables. Working on that.